# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1468867340 25200
#      Mon Jul 18 11:42:20 2016 -0700
# Node ID ed6104a78ebf6b293760a4da281374d53ca664f7
# Parent  18eae417ffec21ff42540a62bb295ae729f67d93
kbhit: return an empty string if not running interactively (bug #48541)

* sysdep.cc (Fkbhit): Return an empty string if not running interactively.

diff --git a/libinterp/corefcn/sysdep.cc b/libinterp/corefcn/sysdep.cc
--- a/libinterp/corefcn/sysdep.cc
+++ b/libinterp/corefcn/sysdep.cc
@@ -715,17 +715,17 @@ x = kbhit (1);
 @end example
 
 @noindent
 is identical to the above example, but doesn't wait for a keypress,
 returning the empty string if no key is available.
 @seealso{input, pause}
 @end deftypefn */)
 {
-  octave_value retval;
+  octave_value retval = "";
 
   // FIXME: add timeout and default value args?
 
   if (octave::application::interactive ())
     {
       Fdrawnow ();
 
       int c = octave_kbhit (args.length () == 0);
